### PR TITLE
Revert move of index_dev.rst to development/index.rst

### PR DESCRIPTION
**Pull request by taldcroft**
### Description

In #16561, @pllim had presciently wondered if there was a problem moving `docs/index_dev.rst` to `docs/development/index.rst`.  Turns out there is an excellent reason, which is that the docs for the stable branch and previous versions on RTD point to a top-level `index_dev.html`. So currently the new dev docs are inaccessible and the `Contributors Guide` link is broken. 

This PR fixes that.

:fire:.

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

**Comment by astrofrog** [comment]
We could always add a redirect if we did want to ever move it?

**Comment by taldcroft** [comment]
> We could always add a redirect if we did want to ever move it?

That was my first thought. I just don't know how to do that. Googling showed https://pypi.org/project/sphinx-reredirects/ and I wasn't sure if we want to go there. 

In general I much prefer the standard layout like `development/index.html`, so if a redirect is possible then I'd be happy to do that instead of this PR.

**Comment by pllim** [comment]
Redirect is hardcoded within the bowels of RTD admin page. If you rather do a redirect, I can help. But we can also consider this PR. Please let me know.

**Comment by taldcroft** [comment]
> Redirect is hardcoded within the bowels of RTD admin page. If you rather do a redirect, I can help. But we can also consider this PR. Please let me know.

I think that having the docs depend on a hidden RTD config seems not so nice. Unless anyone else feels strongly I'd say to just go with this PR.

File: docs/index_dev.rst [hunk 1]
```diff
@@ -58,22 +58,22 @@ Details
...
+   development/maintainers/index
 
 .. Note:: Parts of this guide were adapted from the
     `pandas developer documentation <https://pandas.pydata.org/pandas-docs/stable/development/index.html>`_. Astropy is grateful to the pandas team for their documentation efforts.
 
 {%else%}
 
 To read the developer documentation, you will need to go to the
-`latest developer version of the documentation <https://docs.astropy.org/en/latest/development/index.html>`_.
+`latest developer version of the documentation <https://docs.astropy.org/en/latest/dev_index.html>`_.
```
Comment (pllim) [inline docs/index_dev.rst:77]: Typo?
```suggestion
`latest developer version of the documentation <https://docs.astropy.org/en/latest/index_dev.html>`_.
```

**Review by pllim — APPROVED** [review]
Thanks! Given you prefer to revert via PR, I am going to merge.

**Comment by pllim** [comment]
Thank YOU for doing the actual fix! 😸

**Pull request merged**